Add custom font and scripts notice (#51)
LycheeOrg/LycheeOrg.github.io: docs/faq.md

}
```

### How to use a custom font on the landing page?

> For example, if you want to change it to Roboto, add the following custom CSS to the your `user.css` or via the settings menu:

Reviewer: To me the flow of thought and the introduction feels slightly strange. The initial sentence starts with a specific example, but then the CSS has a placeholder which is generic. Either make it specific or generic, but be consistent about it. Personally, I would start with a generic example.

```suggestion
In order to use a specific font on the landing page which does not rely on some font provided by the client-side browser, you have two options:

 1. self-host the necessary set of font files as part of your Lychee installation
 2. use an external font-hosting service such as Google Fonts

Both options require to tweak the custom `user.css` via the settings menu.

#### Option 1: Self-hosted font files

Let's assume that your favorite font is called `'My Font'`. You need a set of the font files in WOFF2 format with one font file per used font weight and style. The landing page uses 5 different font weights (200 = thin, 300 = light, 400 = normal, 500 = thick, 700 = bold) all in normal (i.e. upright) style.

The necessary CSS consists of two parts: firstly, the font face and its different weights are declared via the `@font-face`-directive; secondly, the declared font needs to be used by the styles for the page. A template CSS code for `user.css` looks like this.
```

```css
@font-face {

>     font-family: "Roboto";

Reviewer: ```suggestion
    font-family: "My Font";
```

    font-style: normal;
    font-weight: 200;

>     src: local(""), url("<font URL for weigth 200>") format("woff2");

Reviewer: ```suggestion
    src: local(""), url("<font URL for weight 200>") format("woff2");
```

}
@font-face {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 300;
    src: local(""), url("<font URL for weigth 300>") format("woff2");
}
@font-face {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 400;
    src: local(""), url("<font URL for weigth 400>") format("woff2");
}
@font-face {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 500;
    src: local(""), url("<font URL for weigth 500>") format("woff2");
}
@font-face {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 700;
    src: local(""), url("<font URL for weigth 700>") format("woff2");
}

#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {

>     font-family: "Roboto", sans-serif !important;

Reviewer: ```suggestion
    font-family: "My Font", sans-serif !important;
```

}

#intro .subtitle,
#logo .subtitle {
    margin-top: 0 !important;

> }

Reviewer: Is this required or related to using an alternative font?

Reviewer: I assumed so. But then I recommend to keep it out of the generic template and present it as part of the template for Google Roboto. Also add a remark about it which explains it.

Reviewer: I'm fine with it.

```

> Probably, you have to customize this:
> * Replace the placeholders with your actual URLs.
> * To use a different font, just change the URLs to match your font and the word `Roboto` to your font's name.
> * If you want to use a hosted service like Google Fonts, replace the `@font-face` blocks with `@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");` and change `Roboto` to your font's name on Google Fonts.

Reviewer: I don't understand the term "probably". It is for sure that the placeholders in the template must be replaced.

```suggestion

#### Option 2: Using an external font-hosting service

If you want to use an external font-hosting service, replace the `@font-face` blocks of the template above with an `@import`-directive which points to the externally hosted font. The name of the font-family must be replaced by the font name as documented by the external font-service.

For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

```css
@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");

#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "Roboto", sans-serif !important;
}
```

```

Sorry, my suggestion misses the final three backticks, because Github becomes confused with them.

    src: local(""), url("<font URL for weight 400>") format("woff2");
}
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 500;
    src: local(""), url("<font URL for weight 500>") format("woff2");
}
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 700;
    src: local(""), url("<font URL for weight 700>") format("woff2");
}

#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "My Font", sans-serif !important;
}
```

#### Option 2: Using an external font-hosting service

If you want to use an external font-hosting service, replace the `@font-face` blocks of the template above with an `@import`-directive which points to the externally hosted font. The name of the font-family must be replaced by the font name as documented by the external font-service.
For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

```css
@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");
#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "Roboto", sans-serif !important;
}

> ```

Reviewer 2: The example is useful but the details need to be updated since we are back to using Roboto in the landing page. Maybe simply give an example of using another Google font (like the one that we used in error pages, whatever it's name was?).

```css
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 200;
    src: local(""), url("<font URL for weight 200>") format("woff2");
}
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 300;
    src: local(""), url("<font URL for weight 300>") format("woff2");
}
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 400;
    src: local(""), url("<font URL for weight 400>") format("woff2");
}
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 500;
    src: local(""), url("<font URL for weight 500>") format("woff2");
}
@font-face {
    font-family: "My Font";
    font-style: normal;
    font-weight: 700;
    src: local(""), url("<font URL for weight 700>") format("woff2");
}

#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "My Font", sans-serif !important;
}

#intro .subtitle,
#logo .subtitle {
    margin-top: 0 !important;
}
```

> #### Option 2: Using an external font-hosting service

Reviewer: I am not 100% sure, but I believe we need a blank line after the caption otherwise the rendering fails. (Of course it ways me how introduced that error due to my suggestion.)

```suggestion
#### Option 2: Using an external font-hosting service

```

If you want to use an external font-hosting service, replace the `@font-face` blocks of the template above with an `@import`-directive which points to the externally hosted font. The name of the font-family must be replaced by the font name as documented by the external font-service.

> For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

Reviewer: Same think here.

```suggestion
For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

```

}

#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "My Font", sans-serif !important;
}

#intro .subtitle,
#logo .subtitle {
    margin-top: 0 !important;
}
```

#### Option 2: Using an external font-hosting service

If you want to use an external font-hosting service, replace the `@font-face` blocks of the template above with an `@import`-directive which points to the externally hosted font. The name of the font-family must be replaced by the font name as documented by the external font-service.
For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

```css
@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");
#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "Roboto", sans-serif !important;
}

#intro .subtitle,
#logo .subtitle {
    margin-top: 0 !important;
}
```

Some fonts may require additional changes to improve the margin of the subtitles. One of these fonts is Roboto. Therefore, you can add following:

```css
#intro .subtitle,
#logo .subtitle {
    margin-top: 0 !important;
}

> ```

Reviewer 2: This CSS section occurs twice before the explanation provided here; as such the explanation seems too late. Can we move it to the discussion of option 1?

### How to add custom scripts?

If you want to add custom scripts, for example to add an analytics service, you can add the script code to the `additional_footer_text` setting. For example, adding `<script>alert('Hello World!');</script>` to it will display an alert dialog with `Hello World!` in it every time you load the page.

> 

Reviewer 2: Why is this part of this PR?
